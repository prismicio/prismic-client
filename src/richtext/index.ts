export { asTree } from "./asTree";
export { asText } from "./asText";

export { serialize } from "./serialize";
export { wrapMapSerializer } from "./wrapMapSerializer";
export { composeSerializers } from "./composeSerializers";

export { RichTextNodeType as Element } from "../types/value/richText";

export type {
	RichTextFunctionSerializer,
	RichTextMapSerializer,
	RichTextMapSerializerFunction,
} from "./types";
