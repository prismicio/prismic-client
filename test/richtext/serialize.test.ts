import { expect, it } from "vitest"

import { createRichTextFixtures } from "../__testutils__/createRichTextFixtures"

import { htmlRichTextFunctionSerializer } from "../__fixtures__/htmlRichTextFunctionSerializer"
import { htmlRichTextMapSerializer } from "../__fixtures__/htmlRichTextMapSerializer"

import type { RichTextFunctionSerializer } from "../../src/richtext"
import { Element, serialize, wrapMapSerializer } from "../../src/richtext"

const serializeMacro =
	(richTextKey: keyof ReturnType<typeof createRichTextFixtures>) => () => {
		const richTextFixtures = createRichTextFixtures()
		const richTextFixture = richTextFixtures[richTextKey]

		const functionSerialization = serialize(
			richTextFixture,
			htmlRichTextFunctionSerializer,
		)
		const mapSerialization = serialize(
			richTextFixture,
			wrapMapSerializer(htmlRichTextMapSerializer),
		)

		expect(functionSerialization).toMatchSnapshot()
		expect(mapSerialization).toMatchSnapshot()
	}

it(
	"serializes a rich text field value using given serializers",
	serializeMacro("en"),
)

it("handles Arabic characters correctly", serializeMacro("ar"))

it("handles Chinese characters correctly", serializeMacro("cn"))

it("handles Korean characters correctly", serializeMacro("ko"))

it("handles emoji characters correctly", serializeMacro("emoji"))

// See: https://github.com/prismicio/prismic-client/issues/198
it("handles overlapped styling correctly", serializeMacro("overlapped"))

it("omits nullish serialized values from the result", () => {
	const richTextFixtures = createRichTextFixtures()

	// We are expecting only heading1 to be included in the serialized result.
	// Note that we are returning `null` for embed. This will be omitted just like
	// the other `undefined` return values.
	const serializer: RichTextFunctionSerializer<string> = (
		_type,
		node,
		text,
		_children,
		_key,
	) => {
		switch (node.type) {
			case Element.heading1: {
				return `<h1>${text}</h1>`
			}

			case Element.embed: {
				return null
			}
		}
	}

	const serialization = serialize(richTextFixtures.en, serializer)

	expect(serialization).toMatchSnapshot()
})
