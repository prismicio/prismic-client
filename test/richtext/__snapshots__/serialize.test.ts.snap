// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`handles Arabic characters correctly 1`] = `
[
  "<p> الم<strong>عروف<em> من</em>ذ </strong>زمن طو<span class="label_1">يل أ</span>ن المحتوى القابل للقراءة لصفحة ما سيشتت انتباه القارئ عند النظر إلى شكلها الخارجي. والهدف من </p>",
  "<ol><li>طبيعي إلى حد ما للأحرف، على</li><li>ر المكتبي ومحرري صفحات الوي</li><li>حث عن نص لوريم إيبسوم إلى</li></ol>",
  "<p>إيبسوم إلى اكتشاف العديد من ال<a href="linkResolver(YCiUyRUAACQAxfZK)">مواقع الإلك</a>ترونية التي لا تزال في مهدها. تطورت إصدارات مختلفة على مر السنين، أحيانًا عن طريق الصدفة، وأحيانًا عن عمد (إدخال الفكاهة وما شابه ذلك).</p>",
]
`;

exports[`handles Arabic characters correctly 2`] = `
[
  "<p> الم<strong>عروف<em> من</em>ذ </strong>زمن طو<span class="label_1">يل أ</span>ن المحتوى القابل للقراءة لصفحة ما سيشتت انتباه القارئ عند النظر إلى شكلها الخارجي. والهدف من </p>",
  "<ol><li>طبيعي إلى حد ما للأحرف، على</li><li>ر المكتبي ومحرري صفحات الوي</li><li>حث عن نص لوريم إيبسوم إلى</li></ol>",
  "<p>إيبسوم إلى اكتشاف العديد من ال<a href="linkResolver(YCiUyRUAACQAxfZK)">مواقع الإلك</a>ترونية التي لا تزال في مهدها. تطورت إصدارات مختلفة على مر السنين، أحيانًا عن طريق الصدفة، وأحيانًا عن عمد (إدخال الفكاهة وما شابه ذلك).</p>",
]
`;

exports[`handles Chinese characters correctly 1`] = `
[
  "<p>得辞<strong>象今歓<em>予板</em>聞所詐上</strong>修戸間。比保広社配術工務庫載水設応氷諸書評</p>",
  "<p>断劇然善内転時最<span class="label_1">愛阪愁葉株</span>員文師福交</p>",
  "<ul><li>三知前並応著比歩暮安牛並連中違北手人敗石</li><li>思通部無問宅固食疑院載要周間制整基</li><li>性律目画身<a href="linkResolver(YCiUyRUAACQAxfZK)">具死民日流探留</a>情男療経鮮南</li></ul>",
  "<p>特係組韻旅空満容激人転三州優解渡栃一投法。級森決読策挙際無国相接三理変住忠再野。</p>",
]
`;

exports[`handles Chinese characters correctly 2`] = `
[
  "<p>得辞<strong>象今歓<em>予板</em>聞所詐上</strong>修戸間。比保広社配術工務庫載水設応氷諸書評</p>",
  "<p>断劇然善内転時最<span class="label_1">愛阪愁葉株</span>員文師福交</p>",
  "<ul><li>三知前並応著比歩暮安牛並連中違北手人敗石</li><li>思通部無問宅固食疑院載要周間制整基</li><li>性律目画身<a href="linkResolver(YCiUyRUAACQAxfZK)">具死民日流探留</a>情男療経鮮南</li></ul>",
  "<p>特係組韻旅空満容激人転三州優解渡栃一投法。級森決読策挙際無国相接三理変住忠再野。</p>",
]
`;

exports[`handles Korean characters correctly 1`] = `
[
  "<p>국가는 <strong>건전한 <em>소비행</em>위를</strong> 계도하고 <span class="label_1">생산품의</span> 품질향상을 촉구하기</p>",
  "<ol><li>위한 소비자보호운동을</li><li>법률이 정하는</li><li>바에 의하여 보장한다.</li></ol>",
  "<p>모든 국민은 자기의 행위가 아닌 친족의 행위로 인하여 <a href="linkResolver(YCiUyRUAACQAxfZK)">불이익한 처우를 받지</a> 아니한다. 국민에 대하여 책임을 진다. 감사위원은 원장의 제청으로 대통령이 임명하고.</p>",
]
`;

exports[`handles Korean characters correctly 2`] = `
[
  "<p>국가는 <strong>건전한 <em>소비행</em>위를</strong> 계도하고 <span class="label_1">생산품의</span> 품질향상을 촉구하기</p>",
  "<ol><li>위한 소비자보호운동을</li><li>법률이 정하는</li><li>바에 의하여 보장한다.</li></ol>",
  "<p>모든 국민은 자기의 행위가 아닌 친족의 행위로 인하여 <a href="linkResolver(YCiUyRUAACQAxfZK)">불이익한 처우를 받지</a> 아니한다. 국민에 대하여 책임을 진다. 감사위원은 원장의 제청으로 대통령이 임명하고.</p>",
]
`;

exports[`handles emoji characters correctly 1`] = `
[
  "<p>🔬📗👉<strong>🐬🔱📀 <em>💰🍣</em>💵💩</strong>🔅 🍥🌰💸📮🔳🕀 💁🍧🌅👅💽</p>",
  "<ul><li>🏂 📛🏫🕓📄🌈 🐄💹👟🌓👱🐐👊 🍲👾👅👌🐠🐺 👡🔂📬🔤👦👠 🍊</li><li>🍗💢📯 🍵🔟📗👳🌲 🌝🌄🍓</li><li>🐤 📝💀🔓🐔🔡. 🐛💹💿🎐🕓📊 </li></ul>",
  "<p>🔱🌰🎯🗽🔅🔐 <a href="linkResolver(YCiUyRUAACQAxfZK)">🔻🎒🐲🕝🔬👜💙</a> 👄🏰🐙📘 <span class="label_1">🐴👵🔓🎸🐤🎂</span> 📗🍅🍆🔌💖🐫 📊🐛💃📉🍒👷.</p>",
]
`;

exports[`handles emoji characters correctly 2`] = `
[
  "<p>🔬📗👉<strong>🐬🔱📀 <em>💰🍣</em>💵💩</strong>🔅 🍥🌰💸📮🔳🕀 💁🍧🌅👅💽</p>",
  "<ul><li>🏂 📛🏫🕓📄🌈 🐄💹👟🌓👱🐐👊 🍲👾👅👌🐠🐺 👡🔂📬🔤👦👠 🍊</li><li>🍗💢📯 🍵🔟📗👳🌲 🌝🌄🍓</li><li>🐤 📝💀🔓🐔🔡. 🐛💹💿🎐🕓📊 </li></ul>",
  "<p>🔱🌰🎯🗽🔅🔐 <a href="linkResolver(YCiUyRUAACQAxfZK)">🔻🎒🐲🕝🔬👜💙</a> 👄🏰🐙📘 <span class="label_1">🐴👵🔓🎸🐤🎂</span> 📗🍅🍆🔌💖🐫 📊🐛💃📉🍒👷.</p>",
]
`;

exports[`handles overlapped styling correctly 1`] = `
[
  "<p>Lorem ipsum dolor <strong><em>sit</em> amet</strong></p>",
  "<p>Lorem ipsum dolor <strong><em>sit</em> amet</strong></p>",
  "<p>Lorem ipsum dolor <strong>sit <em>amet</em></strong></p>",
  "<p>Lorem ipsum dolor <strong>sit <em>amet</em></strong></p>",
  "<p>Lorem ipsum dolor <em><strong>sit amet</strong></em></p>",
  "<p>Lorem <em>ipsum <strong>dolor</strong></em><strong> sit</strong> amet</p>",
]
`;

exports[`handles overlapped styling correctly 2`] = `
[
  "<p>Lorem ipsum dolor <strong><em>sit</em> amet</strong></p>",
  "<p>Lorem ipsum dolor <strong><em>sit</em> amet</strong></p>",
  "<p>Lorem ipsum dolor <strong>sit <em>amet</em></strong></p>",
  "<p>Lorem ipsum dolor <strong>sit <em>amet</em></strong></p>",
  "<p>Lorem ipsum dolor <em><strong>sit amet</strong></em></p>",
  "<p>Lorem <em>ipsum <strong>dolor</strong></em><strong> sit</strong> amet</p>",
]
`;

exports[`omits nullish serialized values from the result 1`] = `
[
  "<h1>Lorem ipsum dolor sit amet.</h1>",
  "<h1>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</h1>",
]
`;

exports[`serializes a rich text field value using given serializers 1`] = `
[
  "<h1>Lorem ipsum dolor sit amet.</h1>",
  "<h2>Lorem ipsum dolor sit amet.</h2>",
  "<h3>Lorem ipsum dolor sit amet.</h3>",
  "<h4>Lorem ipsum dolor sit amet.</h4>",
  "<h5>Lorem ipsum dolor sit amet.</h5>",
  "<h6>Lorem ipsum dolor sit amet.</h6>",
  "<p>Lorem ipsum dolor sit amet.</p>",
  "<p></p>",
  "<h1>Lorem ipsum dolor <strong>sit amet</strong>, consectetur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</h1>",
  "<p>Lorem ipsum dolor <strong>sit amet</strong>, consectetur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="testLabel">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>",
  "<ol><li>Lorem ipsum dolor <strong>sit amet</strong>, consectetur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</li><li>Lorem ipsum dolor sit amet.</li><li>Lorem ipsum dolor sit amet.</li></ol>",
  "<ul><li>Lorem ipsum dolor <strong>sit amet</strong>, consectetur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</li><li>Lorem ipsum dolor sit amet.</li><li>Lorem ipsum dolor sit amet.</li></ul>",
  "<p>Lorem ipsum dolor <strong>sit amet</strong>, consectetur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>",
  "<p>Lorem ipsum dolor <strong>sit amet</strong>, consectetur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui offic<span class="testLabel">ia deserunt mollit anim id est laborum.</span></p>",
  "<img src="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format" alt="An Atlantic Puffin" />",
  "<img src="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format" alt="An Atlantic Puffin" />",
  "<p><span class="testLabel">Lorem ipsum dolor <strong>sit amet</strong>, consecte</span>tur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>",
  "<img src="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format" alt="An Atlantic Puffin" />",
  "<p><span class="label">Lorem ipsum dolor <strong>sit amet</strong>, consecte</span>tur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>",
  "<pre>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</pre>",
  "<p><span class="label">Lorem ipsum dolor <strong>sit amet</strong>, consecte</span>tur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>",
  "<iframe width="200" height="113" src="https://www.youtube.com/embed/n5CwXuyNfoc?feature=oembed" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>",
  "<p><span class="label">Lorem ipsum dolor <strong>sit amet</strong>, consecte</span>tur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>",
  "<img src="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format" alt="An Atlantic Puffin" />",
]
`;

exports[`serializes a rich text field value using given serializers 2`] = `
[
  "<h1>Lorem ipsum dolor sit amet.</h1>",
  "<h2>Lorem ipsum dolor sit amet.</h2>",
  "<h3>Lorem ipsum dolor sit amet.</h3>",
  "<h4>Lorem ipsum dolor sit amet.</h4>",
  "<h5>Lorem ipsum dolor sit amet.</h5>",
  "<h6>Lorem ipsum dolor sit amet.</h6>",
  "<p>Lorem ipsum dolor sit amet.</p>",
  "<p></p>",
  "<h1>Lorem ipsum dolor <strong>sit amet</strong>, consectetur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</h1>",
  "<p>Lorem ipsum dolor <strong>sit amet</strong>, consectetur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="testLabel">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>",
  "<ol><li>Lorem ipsum dolor <strong>sit amet</strong>, consectetur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</li><li>Lorem ipsum dolor sit amet.</li><li>Lorem ipsum dolor sit amet.</li></ol>",
  "<ul><li>Lorem ipsum dolor <strong>sit amet</strong>, consectetur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</li><li>Lorem ipsum dolor sit amet.</li><li>Lorem ipsum dolor sit amet.</li></ul>",
  "<p>Lorem ipsum dolor <strong>sit amet</strong>, consectetur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>",
  "<p>Lorem ipsum dolor <strong>sit amet</strong>, consectetur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui offic<span class="testLabel">ia deserunt mollit anim id est laborum.</span></p>",
  "<img src="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format" alt="An Atlantic Puffin" />",
  "<img src="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format" alt="An Atlantic Puffin" />",
  "<p><span class="testLabel">Lorem ipsum dolor <strong>sit amet</strong>, consecte</span>tur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>",
  "<img src="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format" alt="An Atlantic Puffin" />",
  "<p><span class="label">Lorem ipsum dolor <strong>sit amet</strong>, consecte</span>tur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>",
  "<pre>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</pre>",
  "<p><span class="label">Lorem ipsum dolor <strong>sit amet</strong>, consecte</span>tur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>",
  "<iframe width="200" height="113" src="https://www.youtube.com/embed/n5CwXuyNfoc?feature=oembed" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>",
  "<p><span class="label">Lorem ipsum dolor <strong>sit amet</strong>, consecte</span>tur <em>adipiscing elit</em>, sed do eiusmod tempor incididunt ut labore et dolore <span class="label">magna aliqua</span>. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea <a href="https://google.com" target="undefined">commodo consequat</a>. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat <a href="linkResolver(XvoFFREAAM0WGBng)">nulla pariatur</a>. Excepteur sint occaecat cupidatat <a href="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format">non proident</a>, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>",
  "<img src="https://images.prismic.io/200629-sms-hoy/f0a757f6-770d-4eb8-a08b-f1727f1a58e4_guilherme-romano-KI2KaOeT670-unsplash.jpg?auto=compress,format" alt="An Atlantic Puffin" />",
]
`;
